dist/ethereum.js

```diff
@@ -2604,39 +2615,65 @@ var requestManager = function() {
         }
 
         // HTTP ASYNC (only when callback is given, and it a HttpProvidor)
-        if(typeof callback === 'function' && provider.name === 'HTTP'){
+        if(typeof callback === 'function' && provider.name === 'RPC over HTTP'){
```

Comment:
The problem here is that only HTTP has currently callback style responses, all others are return values. Other providers could use callbacks later too, and we just add those to the list, but we won't need to introduce more complexity anyway.